extensions/standard-processors/tests/unit/PutUDPTests.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <new>
+#include <random>
+#include <string>
+#include <unordered_map>
+#include "range/v3/view/transform.hpp"
+#include "range/v3/range/conversion.hpp"
+#include "TestBase.h"
+#include "PutUDP.h"
+#include "utils/net/DNS.h"
+#include "utils/net/Socket.h"
+#include "utils/OptionalUtils.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+namespace {
+class SingleInputTestController : public TestController {
```

Comment:
I really like this and I feel this could be useful in the future. Could you move this to TestBase.h for easier use in other PR-s aswell?

Reply:
I think it's better if we move it in the first reuse, and I would rather move it into its own header. I can convert RouteText to use this if you and @adamdebreceni (author of RouteText) think it would be useful, and that would be a first reuse.

Reply:
I agree that using this class could greatly simplify many test cases that use `GetFile`/`PutFile` to read/write flowfiles while testing some other processor (`ListenHTTPTests`, `ExpressionLanguageTests`, `PutSFTPTests`, `ExtractTextTests`, `FocusArchiveTests`, `TensorFlowTests` seem to be the ones in this category), the straightforward cases could fit into this PR (maybe `RouteTextTests`) but we could do them in a separate PR(s)

Reply:
Ok, then I'll extract the common parts from this and `RouteTextController` and leave the rest for future refactoring efforts.

Reply:
In the end I spent a few hours on `RouteTextController` and saw that performing this would require a rewrite of the tests, which I don't want to do.